dashboard/impl/src/java/org/sakaiproject/dash/logic/DashboardUserLogicImpl.java

```diff
@@ -108,19 +111,88 @@ public int countNewsLinksByGroupId(String sakaiUserId,
 							item.setTitle(typeObj.getGroupTitle(itemCount, item.getContext().getContextTitle(), item.getNewsTimeLabelKey()));
 						}
 					}
+				}else{
+					//When getItemCount() > 1 the infoLinkUrl is null and hence we don't need to run this call.
+					getInfoLinkUrlForNewsItem(item);
 				}
 			}
 		}
-		
+
 		return links;
 	}
 
+	private void getInfoLinkUrlForNewsItem(NewsItem item) {
+		if(item!=null){
+			SourceType sourceType = item.getSourceType();
+			if(sourceType != null) {
+				DashboardEntityInfo typeObj = dashboardLogic.getDashboardEntityInfo(sourceType.getIdentifier());
+				if(typeObj != null) {
+					//getValues() take one of the parameter as Locale code and this used for logging warn messages
+					//so sending null should be fine here.
+					Map<String, Object> values = typeObj.getValues(item.getEntityReference(), null);
+					List<Map<String,String>> moreInfo=(ArrayList<Map<String, String>>) values.get(DashboardEntityInfo.VALUE_MORE_INFO);
+					if(moreInfo!=null){
+						for (Map<String, String> map : moreInfo) {
+							item.setInfoLinkURL(map.get(DashboardEntityInfo.VALUE_INFO_LINK_URL));
+
+						}
+					}else{
+						logger.debug("more.Info is null");
+					}
+				}else{
+					logger.debug("DashboardEntityInfo is null");
+				}
+			}else{
+				logger.debug("SourceType like is null");
+			}
+
+		}else{
+			logger.debug("NewsItem is null");
+		}
+		
+	}
+	private void getInfoLinkUrlForCalendarItem(CalendarItem item) {
+		if(item!=null){
+			SourceType sourceType = item.getSourceType();
+			if(sourceType != null) {
+				DashboardEntityInfo typeObj = dashboardLogic.getDashboardEntityInfo(sourceType.getIdentifier());
+				if(typeObj != null) {
+					//getValues() take one of the parameter as Locale code and this used for logging warn messages
+					//so sending null should be fine here.
+					Map<String, Object> values = typeObj.getValues(item.getEntityReference(), null);
+					List<Map<String,String>> moreInfo=(ArrayList<Map<String, String>>) values.get(DashboardEntityInfo.VALUE_MORE_INFO);
```

Comment:
Why the moreInfo is of type List<Map<String, String>> here? It is not intuitive. Could you please double check?